### PR TITLE
Fix Trainable Tokens output-head bias

### LINKED ISSUE #3649: Trainable Tokens drops the output-head bias before merge
### System Info

PEFT `main` at `3d881e97426db449902478fa545ed6fa3086991f` (`0.20.1.dev0`), Python 3.12.13, PyTorch 2.13.0+cpu, Transformers 5.14.1, Linux/WSL2 CPU.

### Reproduction

Trainable Tokens omits an existing output-head bias in its unmerged Linear forward path. On a pretrained BERT model, adding a fresh adapter changes the logits without any training, while disabling or merging it restores the original output. Both standalone Trainable Tokens and LoRA's `trainable_token_indices` are affected.

```python
import copy

import torch
from transformers import BertForMaskedLM, BertTokenizer
from peft import LoraConfig, TrainableTokensConfig, get_peft_model

model_id = "prajjwal1/bert-tiny"
revision = "6f75de8b60a9f8a2fdf7b69cbd86d9e64bcb3837"
base = BertForMaskedLM.from_pretrained(model_id, revision=revision).eval()
tokenizer = BertTokenizer.from_pretrained(model_id, revision=revision)
inputs = tokenizer("The capital of France is [MASK].", return_tensors="pt")
with torch.no_grad():
    expected = base(**inputs).logits
bias = base.get_output_embeddings().bias.detach().clone()

configs = [
    TrainableTokensConfig(token_indices=[tokenizer.mask_token_id]),
    LoraConfig(r=2, target_modules=["query"],
               trainable_token_indices=[tokenizer.mask_token_id]),
]
for config in configs:
    model = get_peft_model(copy.deepcopy(base), config).eval()
    with torch.no_grad():
        active = model(**inputs).logits
        with model.disable_adapter():
            disabled = model(**inputs).logits
        model.merge_adapter()
        merged = model(**inputs).logits
    print(type(config).__name__)
    print("active error:", (active - expected).abs().max().item())
    print("active + bias error:", (active + bias - expected).abs().max().item())
    print("disabled error:", (disabled - expected).abs().max().item())
    print("merged error:", (merged - expected).abs().max().item())
```

Both configurations produce:

```text
active error: 9.51423454284668
active + bias error: 0.0
disabled error: 0.0
merged error: 0.0
```

No training or replacement of pretrained parameters is involved. The explicit BERT classes avoid model-type inference for this older checkpoint. Loading reports unrelated NSP/pooler keys as unexpected, not missing decoder weights.

The [Linear branch in TrainableTokensLayer](https://github.com/huggingface/peft/blob/3d881e97426db449902478fa545ed6fa3086991f/src/peft/tuners/trainable_tokens/layer.py#L263-L268) calls `F.linear(input=x, weight=W)` without the base layer's bias. The merged and disabled paths call the original layer, which includes it.

### Expected behavior

With default initialization, a fresh adapter should preserve the original model output. Active and merged inference should agree, including the existing output-head bias, without changing which bias parameters are trainable.

### Proposed scope and ownership

I would like to take ownership of this fix. Would it be okay for me to submit a focused PR preserving the Linear bias and adding regression coverage for both standalone Trainable Tokens and the LoRA auxiliary path? The tests can use a tiny locally initialized BERT with nonzero-bias and zero-bias controls, so CI would not need to download this checkpoint.

I did not find an overlapping issue or open PR; #2863 fixed Linear initialization rather than this forward-path behavior. I will wait for maintainer confirmation before implementing the patch.

### PR DESCRIPTION
**Pull request by DaoyuanLi2816**
## Summary

Trainable Tokens uses a functional Linear forward while an adapter is active. That path preserved the updated weight but omitted the wrapped output head's bias, so fresh adapters changed logits before merge.

Pass the existing bias to `F.linear` and add local regression coverage for both standalone Trainable Tokens and LoRA `trainable_token_indices`, including active, disabled, and merged states.

Closes #3649.

Coordination: #3649 contains the reproducer and proposed fix. BenjaminBossan confirmed that the original reporter has priority to implement it in https://github.com/huggingface/peft/pull/3651#issuecomment-5523292539.

## Tests

- `python -m pytest tests/test_trainable_tokens.py -q --no-cov --regression` — 62 passed
- `python -m pytest tests -k trainable_tokens -q --no-cov --regression` — 186 passed, 13 skipped; two unrelated existing CUDA `EmbConv1D` failures also reproduce on unchanged `origin/main`
- `uvx --from ruff==0.16.4 ruff check src/peft/tuners/trainable_tokens/layer.py tests/test_trainable_tokens.py`
- `uvx --from ruff==0.16.4 ruff format --check src/peft/tuners/trainable_tokens/layer.py tests/test_trainable_tokens.py`
- `git diff --check`

AI assistance was used. I reviewed every changed line, ran the tests above, understand the change end-to-end, and take responsibility for the contribution.

### CODE REVIEW

File: tests/test_trainable_tokens.py [hunk 1]
```diff
@@ -80,6 +80,16 @@ def forward(self, x):
         return self.lin0(self.embed_in(x))
 
 
+class ModelWithOutputHead(torch.nn.Module):
```
Comment (githubnemo) [inline tests/test_trainable_tokens.py]:
```suggestion
class ModelWithOutputHeadBias(torch.nn.Module):
```

Let's make this explicit

**Review by githubnemo — COMMENTED** [review]
One nit, otherwise LGTM

**Comment by DaoyuanLi2816** [comment]
@githubnemo Could you rerun the [Windows/Python 3.13 job](https://github.com/huggingface/peft/actions/runs/34151979612/job/106316208442)? My rerun request was denied for lack of repository admin rights. Its failures are BF16 primitive-creation errors and worker crashes with `0xc000001d`; no Trainable Tokens test appears in the failure summary. The other seven matrix jobs passed. [The main-branch Windows/Python 3.13 job](https://github.com/huggingface/peft/actions/runs/35620469525/job/106415194416) passed with the same reported runner image, Python and PyTorch versions. That is not a controlled baseline for this PR, so I have not treated the failure as proven unrelated.

File: tests/test_trainable_tokens.py (hunk 1 above)
Comment (DaoyuanLi2816) [reply tests/test_trainable_tokens.py]: @githubnemo Renamed the fixture and its use to ModelWithOutputHeadBias in 9f334abf. The full tests/test_trainable_tokens.py suite passes with --regression (62 passed), and Ruff 0.16.4 check/format and git diff --check pass. No production-code changes in this update.

**Review by githubnemo — APPROVED** [review]
LGTM, thanks :)
